userspace/falco/app_actions/configure_interesting_sets.cpp

```diff
@@ -15,27 +15,51 @@ limitations under the License.
 */
 
 #include "application.h"
+#include "configure_interesting_sets.h"
+#include <unordered_set>
+#include <sinsp.h>
+#include <sstream>
 
 using namespace falco::app;
 
-void application::configure_interesting_sets()
+extern sinsp_evttables g_infotables;
+
+std::string concat_names(std::unordered_set<std::string> const names)
 {
-	/// TODO: in the next future we need to change the interface of `enforce_simple_ppm_sc_set`
-	/// and `enforce_sinsp_state_tp` APIs, they shouldn't require an inspector to be called!
-	std::unique_ptr<sinsp> inspector(new sinsp());
+	std::set<std::string> names_ordered = {};
+	for (const auto &n : names)
+	{
+		names_ordered.insert(n);
+	}
+	std::stringstream ss;
+	std::copy(names_ordered.begin(), names_ordered.end(),
+	std::ostream_iterator<std::string>(ss, ", "));
+	std::string names_str = ss.str();
+	return names_str.substr(0, names_str.size() - 2);
+}
 
-	/* Please note: here we fill these 2 sets because we are interested in only some features, if we leave
-	 * them empty `libsinsp` will fill them with all the available syscalls and all the available tracepoints!
-	 */
+void application::extract_rules_event_names(std::unique_ptr<sinsp>& inspector, std::unordered_set<std::string>& rules_evttypes_names)
+{
+	/* Get all (positive) PPME events from all rules as idx codes.
+	 * Events names from negative filter expression statements are NOT included.
+	 * PPME events in libsinsp are needed to map each event type into it's enter and exit event if applicable (e.g. for syscall events).
+	*/
+	std::set<uint16_t> rule_events;
+	std::string source = falco_common::syscall_source;
+	m_state->engine->evttypes_for_ruleset(source, rule_events);
+	std::unordered_set<uint32_t> ppme_events_codes(rule_events.begin(), rule_events.end());
 
-	/* Here the `libsinsp` state set is not enough, we need other syscalls used in the rules,
-	 * so we use the `simple_set`, this `simple_set` contains all the syscalls of the `libsinsp` state
-	 * plus syscalls for Falco default rules.
-	 */
-	m_state->ppm_sc_of_interest = inspector->enforce_simple_ppm_sc_set();
+	/* Translate PPME event idx codes to consolidated event names.
+	 * Those are the exact event type (evt.type) names from the rules and hence also contain non syscall names, e.g. "container".
+	*/
+	rules_evttypes_names = inspector->get_events_names(ppme_events_codes);
+}
+
+void application::activate_interesting_events(std::unique_ptr<sinsp>& inspector)
```

Comment:
@FedeDP extracted this out and it should go last, in fact, since it was done before the -A check it kind of was a small bug I introduced ... should be fixed now.